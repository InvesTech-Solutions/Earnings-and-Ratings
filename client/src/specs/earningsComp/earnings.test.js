import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Earnings from './earnings.js';

describe('component: Earnings', () => {
  it('should render initial layout', () => {
    let wrapper = shallow(<Earnings />);
    expect(wrapper.getElements()).toMatchSnapshot();
  });

  it('After component updated, it should produce yRange in a $XX.XX format string', () => {
    let wrapper = shallow(<Earnings />);
    setTimeout(() => {
      let yRange = wrapper.state().yRange;
      let formatCheck = yRange.reduce((accum, element) => {
        if (accum === false) { return false; }
        let dollar = element[0] === '$';
        let dot = element.substring(element.length - 3)[0] === '.';
        if (dollar === true || dot === true) {
          return true;
        }
        return false;
      }, true);

      expect(formatCheck).toEqual(true);
    }, 0);
  });
});
