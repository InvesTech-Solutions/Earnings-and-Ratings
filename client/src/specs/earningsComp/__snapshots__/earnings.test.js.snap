// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: Earnings should render initial layout 1`] = `
Array [
  <section
    id="earningsComponent"
  >
    <header
      className="earningsHead"
    >
      Earnings
    </header>
    <div
      className="chartDiv"
    >
      <div
        className="chartContainer"
      >
        <div
          className="innerChartContainer"
        >
          <div
            className="axes"
          >
            <YAxis
              yRange={Array []}
            />
            <DataPoints
              actual={undefined}
              estimate={undefined}
              yRange={Array []}
            />
            <XAxis
              dates={Array []}
            />
          </div>
        </div>
      </div>
    </div>
    <Description
      actual={undefined}
      estimate={undefined}
    />
  </section>,
]
`;
